bundles/org.openhab.binding.freemobilesms/src/main/java/fr/free/smsapi/Sender.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package fr.free.smsapi;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An helper class to compute and send request.
+ *
+ * @author Guilhem Bonnefille - Initial contribution
+ */
+@NonNullByDefault
+public class Sender {
+
+    static public final String DEFAULT_URL = "https://smsapi.free-mobile.fr/sendmsg";
+
+    static final String charset = "UTF-8";
+
+    private final Logger logger = LoggerFactory.getLogger(Sender.class);
+
+    protected final String url;
+
+    public Sender(String url) {
+        this.url = url;
+    }
+
+    public Sender() {
+        this.url = DEFAULT_URL;
+    }
+
+    public String getQuery(Account account, String message) throws UnsupportedEncodingException {
+        String query = String.format("%s?user=%s&pass=%s&msg=%s",
+                this.url,
+                URLEncoder.encode(account.getUser(), charset),
+                URLEncoder.encode(account.getPassword(), charset),
+                URLEncoder.encode(message, charset));
+        return query;
+    }
+
+    public void send(Account account, String message) throws IOException {
+        String urlString = getQuery(account, message);
+        logger.debug("Computed request: {}", urlString);
+        URL url = new URL(urlString);
+        send(url);
+    }
+
+    public void send(URL url) throws IOException {
+        HttpsURLConnection connection = null;
```

Comment:
I think most of this could directly be replaced by using ```org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl```

Reply:
Yes use HttpUtil method or use `HttpClientFactory` to get an `HttpClient`. But that is a bit more work, so if `HttpUtil` would work it is the easiest way to go.